python/ml4ir/base/model/architectures/auto_dag_network.py

```diff
@@ -0,0 +1,362 @@
+                                                    not layer_node.is_input_node]
+        self.file_io.logger.info("Execution order: %s", self.execution_order)
+        self.output_node = model_graph.output_node
+
+    def instantiate_op(self, current_layer_type: str, current_layer_args: Dict) -> layers.Layer:
+        """
+        Create and return an instance of `current_layer_type` with `current_layer_args` params
+        If `current_layer_type` is not found in the set of subclasses of keras.layers.Layer, throws KeyError
+
+        Parameters
+        ----------
+        current_layer_type: str
+            Refers to class name inheriting directly or indirectly from keras.layers.Layer to be instantiated
+        current_layer_args: dict
+            All the arguments from model config which are used as kwargs to current_layer_type instance
+
+        Returns
+        -------
+        keras.layers.Layer
+            Instance of current_layer_type
+        """
+        try:
+            return self.available_layers[current_layer_type](**current_layer_args)
+        except KeyError:
+            self.file_io.logger.error("Layer type: '%s' is not supported or not found in %s",
+                                      current_layer_type, str(self.available_layers))
+            raise KeyError(f"Layer type: '{current_layer_type}' "
+                           f"is not supported or not found in subclasses of keras.layers.Layer")
+
+    def get_layer_op(self, layer_args: Dict) -> Dict[str, Union[str, bool, layers.Layer]]:
+        """
+        Define the layer operation for a layer in the model config
+
+        Parameters
+        ----------
+        layer_args: dict
+            All the arguments from model config which are needed to define a layer op
+
+        Returns
+        -------
+        dict
+            Dictionary with the layer instance, inputs required for the layer and the format of inputs to the layer
+        """
+        return {
+            self.INPUTS: layer_args[self.INPUTS],
+            self.OP_IDENTIFIER: self.instantiate_op(layer_args[self.LAYER_TYPE], layer_args.get(self.LAYER_KWARGS, {})),
+            self.INPUTS_AS_LIST: layer_args.get(self.INPUTS_AS_LIST, False)
+        }
+
+    def define_architecture(self, model_config: dict) -> LayerGraph:
+        """
+        Convert the model from model_config to a LayerGraph
+
+        Parameters
+        ----------
+        model_config: dict
+            Model config parsed into a dictionary
+
+        Returns
+        -------
+        LayerGraph
+            Dependency DAG for the given model config
+        """
+
+        layer_ops = {layer_args[self.NAME]: self.get_layer_op(layer_args) for layer_args in
+                     model_config[self.LAYERS_IDENTIFIER]}
+        # While sorting is not mandatory, I would advise not removing it for the sake of reproducibility
+        inputs = sorted(set([input_name for layer_op in layer_ops.values()
```

Comment:
[Minor] @arvindsrikantan Why not sort in `instantiate_op` itself? Seems slightly cleaner.

Reply:
With that, I think you can combine the two methods as well. I don't see why we need a separate method for `get_layer_op` which is essentially a wrapper around `instantiate_op`

Reply:
Good point on combining the two. It was an artifact of "levels" which is not needed anymore

Reply:
> Why not sort in get_layer_op itself? Seems slightly cleaner.

I'm not sorting the inputs to each node here. I'm sorting just the feature layer nodes. The final order is still controlled by the model_config

Reply:
> I'm not sorting the inputs to each node here. I'm sorting just the feature layer nodes.

How are they different?

Reply:
Or rather, why sort then?

Reply:
To keep the order of self.nodes consistent across runs. This also ensures the topological sort result doesn't change every time. Note that there is no unique ordering for topological sort.